dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,115 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+inline void SkipSpace(const char*& ptr) {
+  while(::isspace(*ptr))
+    ptr++;
+}
+
+template <size_t N>
+void Skip(const char*& ptr, const char (&what)[N]) {
+  DALI_ENFORCE(!strncmp(ptr, what, N-1), make_string("Expected \"", what, "\" but got \"", ptr, "\""));
+  ptr += N - 1;
+}
+
+template <size_t N>
+bool TrySkip(const char*& ptr, const char (&what)[N]) {
+  if (!strncmp(ptr, what, N-1)) {
+    ptr += N - 1;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+
+template <size_t N>
+void SkipFieldName(const char*& ptr, const char (&name)[N]) {
+  SkipSpace(ptr);
+  Skip(ptr, "'");
+  Skip(ptr, name);
+  Skip(ptr, "'");
+  SkipSpace(ptr);
+  Skip(ptr, ":");
+  SkipSpace(ptr);
+}
+
+std::string ParseStringValue(const char*& input, char delim_start = '\'', char delim_end = '\'') {
+  DALI_ENFORCE(*input++ == delim_start, make_string("Expected \'", delim_start, "\'"));
+  std::string out;
+  for (;*input != '\0'; input++) {
```

Comment:
Is `input++` safe, when `input` is `const char*&`? It's like iterating the referece, right? IMHO `const char*` would be safer and clearer, but that's your call ;)

Reply:
We want to advance the pointer that was passed to this function. The alternative would be to return the new pointer, but then we'd have no way of returning a value from this function. The input pointer is a minimalistic stream.